src/main/java/teammates/ui/newcontroller/Action.java

```diff
@@ -64,24 +65,26 @@ public void setEmailSender(EmailSender emailSender) {
     /**
      * Returns true if the requesting user has sufficient authority to access the resource.
      */
-    protected boolean checkAccessControl() {
+    protected void checkAccessControl() {
         if (authType.getLevel() < getMinAuthLevel().getLevel()) {
             // Access control level lower than required
-            return false;
+            throw new UnauthorizedAccessException();
         }
 
         if (getMinAuthLevel() == AuthType.PUBLIC) {
             // No authentication necessary for this resource
-            return true;
+            return;
         }
 
         if (authType == AuthType.ALL_ACCESS) {
             // All-access pass granted
-            return true;
+            return;
         }
 
         // All other cases: to be dealt in case-by-case basis
-        return checkSpecificAccessControl();
+        if (!checkSpecificAccessControl()) {
```

Comment:
`checkSpecificAccessControl()` could also throw the `UnauthorizedAccessException` directly?
 P.S. `check*` sounds like a action rather than something returns a boolean.